rct229/rulesets/ashrae9012019/section11/section11rule12.py

```diff
@@ -0,0 +1,156 @@
+                list_path="$.buildings[*]",
+            )
+
+        def create_data(self, context, data):
+            rmd_b = context.BASELINE_0
+            rmd_p = context.PROPOSED
+
+            return {"rmd_b": rmd_b, "rmd_p": rmd_p}
+
+        class BuildingRule(RuleDefinitionListIndexedBase):
+            def __init__(self):
+                super(Section11Rule12.RMDRule.BuildingRule, self).__init__(
+                    rmds_used=produce_ruleset_model_description(
+                        USER=False, BASELINE_0=True, PROPOSED=True
+                    ),
+                    each_rule=Section11Rule12.RMDRule.BuildingRule.BuildingSegmentRule(),
+                    index_rmd=BASELINE_0,
+                    list_path="$.building_segments[*]",
+                )
+
+            def create_data(self, context, data):
+                building_b = context.BASELINE_0
+                rmd_b = data["rmd_b"]
+                is_leap_year_b = data["is_leap_year_b"]
+
+                bldg_open_sch_id_b = getattr_(
+                    building_b, "buildings", "building_open_schedule"
+                )
+
+                bldg_open_sch_b = find_exactly_one_schedule(rmd_b, bldg_open_sch_id_b)[
+                    "hourly_values"
+                ]
+
+                hours_this_year = (
+                    LeapYear.LEAP_YEAR_HOURS
+                    if is_leap_year_b
+                    else LeapYear.REGULAR_YEAR_HOURS
+                )
+
+                return {
+                    "bldg_open_sch_b": bldg_open_sch_b,
+                    "hours_this_year": hours_this_year,
+                }
+
+            def list_filter(self, context_item, data):
+                bldg_open_sch_b = data["bldg_open_sch_b"]
+                hours_this_year = data["hours_this_year"]
+
+                return sum(bldg_open_sch_b) == hours_this_year
+
+            class BuildingSegmentRule(RuleDefinitionListIndexedBase):
+                def __init__(self):
+                    super(
+                        Section11Rule12.RMDRule.BuildingRule.BuildingSegmentRule, self
+                    ).__init__(
+                        rmds_used=produce_ruleset_model_description(
+                            USER=False, BASELINE_0=True, PROPOSED=True
+                        ),
+                        each_rule=Section11Rule12.RMDRule.BuildingRule.BuildingSegmentRule.SWHRule(),
+                        index_rmd=BASELINE_0,
+                        list_path="$.zones[*].spaces[*].service_water_heating_uses[*]",
+                    )
+
+                def create_data(self, context, data):
+                    building_segment_p = context.PROPOSED
+                    rmd_p = data["rmd_p"]
+
+                    service_water_heating_use_ids_list_p = (
+                        get_swh_uses_associated_with_each_building_segment(
+                            rmd_p, building_segment_p["id"]
+                        )
+                    )
```

Comment:
I will move this logic to RMD level. for example, creating a 
```python
swh_uses_associated_with_each_building_segment_p: 
{
    building_segment_id: get_swh_uses_associated_with_each_building_segment(
                            rmd_p, building_segment_p["id"]
                        )
    for building_segment_id in find_all("$.buildings[*].building_segments[*].id, rmd_p
}
```

I can skip this create data here.

Reply:
Good point. Addressed.